kolibri/plugins/management/assets/src/vue/manage-class-page/class-delete-modal.vue

```diff
@@ -68,14 +62,10 @@
         required: true,
       },
     },
-    data() {
-      return {
-        error_message: '',
-      };
-    },
     methods: {
-      submit() {
-        // need delete logic here.
+      classDelete() {
+        this.deleteClass(this.classid);
+        this.$emit('close');
```

Comment:
Let's say page 1 has 2 modals to display, we set `m-1` and `m-2` under state.pageState, then we need mutation `mu-1`, `mu-2` for them respectively.
page 2 has 3 modals to display, we set `m-1` and `m-2` and `m-3` under state.pageState, then we need mutation `mu-1`, `mu-2`, `mu-3` for them respectively.
we ended up with many mutations with very general names.
Also this pattern spread simple logic to multiple locations

A better solution could be that the action returns a promise to the modal and we can dismiss the modal based whether or not the promise is resolved.